### PR TITLE
add WebCodec dequeue events

### PR DESCRIPTION
**Pull request by chrisdavidmills**
### Description

This pull request adds the following four pages:

* AudioDecoder dequeue event
* AudioEncoder dequeue event
* VideoDecoder dequeue event
* VideoEncoder dequeue event

I have also added inheritance information to the parent interface pages because these classes now inherit from `EventTarget`. This isn't in the spec yet, but it makes sense, and the Chrome WebIDL reflects this ([example](https://chromium.googlesource.com/chromium/src/+/700a5d85661641333a7cf8422b13849a3148cf4a%5E!/#F3)).

### Motivation

These are useful new features of the sec, which people will want to learn about.

### Additional details

* https://chromestatus.com/feature/5195706034290688

### CODE REVIEW

File: files/en-us/web/api/audiodecoder/dequeue_event/index.md [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+  - Event
+  - ondequeue
+  - AudioDecoder
+  - Experimental
+browser-compat: api.AudioDecoder.dequeue_event
+---
+
+{{securecontext_header}}{{APIRef("WebCodecs API")}}{{SeeCompatTable}}
+
+The **`dequeue`** event of the {{domxref("AudioDecoder")}} interface fires to signal a decrease in {{domxref("AudioDecoder.decodeQueueSize")}}.
```
Comment (beaufortfrancois) [inline files/en-us/web/api/audiodecoder/dequeue_event/index.md:18]: It would be nice to have `AudioDecoder.decodeQueueSize` defined.

Comment (beaufortfrancois) [reply files/en-us/web/api/audiodecoder/dequeue_event/index.md:18]: https://pr21845.content.dev.mdn.mozit.cloud/en-US/docs/Web/API/AudioDecoder/decodeQueueSize is 404 but https://developer.mozilla.org/en-US/docs/Web/API/AudioDecoder/decodeQueueSize exists.

Comment (chrisdavidmills) [reply files/en-us/web/api/audiodecoder/dequeue_event/index.md:18]: @beaufortfrancois Ah yeah, this page is definitely defined. It looks like only preview URLs for the pages that have been changed  are built; so anything that links from them but has not been changed will 404 in the preview environment.

Comment (chrisdavidmills) [reply files/en-us/web/api/audiodecoder/dequeue_event/index.md:18]: Thanks for taking a look.

Comment (beaufortfrancois) [reply files/en-us/web/api/audiodecoder/dequeue_event/index.md:18]: LGTM!

**Review by rachelandrew — APPROVED** [review]

**Comment by bsmth** [comment]
Thanks @chrisdavidmills 🙌🏻 shall we merge this one or is there anything else outstanding?

**Comment by chrisdavidmills** [comment]
@bsmth that's it; please merge!

**Comment by bsmth** [comment]
Thanks, all 👍🏻

**Comment by chrisdavidmills** [comment]
Cheers @bsmth , much appreciated.

**Pull request merged**